src/mlpack/methods/ann/layer/celu.hpp

```diff
@@ -0,0 +1,203 @@
+   * @param output Resulting output activation.
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(const InputType&& input, OutputType&& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, calculating the function
+   * f(x) by propagating x backwards through f. Using the results from the feed
+   * forward pass.
+   *
+   * @param input The propagated input activation f(x).
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename DataType>
+  void Backward(const DataType&& input, DataType&& gy, DataType&& g);
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the non zero gradient.
+  double const& Alpha() const { return alpha; }
+  //! Modify the non zero gradient.
+  double& Alpha() { return alpha; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  /**
+   * Computes the value of activation function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  double Fn(const double x)
+  {
+    if (x < DBL_MAX)
+    {
+      return (x >= 0) ? x : alpha * (std::exp(x / alpha) - 1);
+    }
+
+     return 1.0;
+  }
+
+  /**
+   * Computes the value of activation function using a dense matrix as input.
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename eT>
+  void Fn(const arma::Mat<eT>& x, arma::Mat<eT>& y)
+  {
+    y.set_size(arma::size(x));
+
+    for (size_t i = 0; i < x.n_elem; i++)
+    {
+      y(i) = Fn(x(i));
+    }
+  }
+
+  /**
+   * Computes the first derivative of the activation function.
+   *
+   * @param x Input data.
+   * @param y Propagated data f(x).
+   * @return f'(x)
+   */
+  double Deriv(const double x, const double y)
+  {
+    return (x >= 0) ? 1 : (y / alpha) + 1;
+  }
+
+  /**
+   * Computes the first derivative of the activation function.
+   *
+   * @param x Input data.
+   * @param y Output activations f(x).
+   * @param z The resulting derivatives.
+   */
+  template<typename InputType, typename OutputType>
+  void Deriv(const InputType& x, OutputType& y)
+  {
+    derivative.set_size(arma::size(x));
+
+    for (size_t i = 0; i < x.n_elem; i++)
+    {
+      derivative(i) = Deriv(x(i), y(i));
+    }
+  }
+
+  //! Locally-stored delta object.
+  OutputDataType delta;
+
+  //! Locally-stored output parameter object.
+  OutputDataType outputParameter;
+
+  //! Locally stored first derivative of the activation function.
+  arma::mat derivative;
+
+  //! CELU Hyperparameter (0 < alpha)
```

Comment:
Picky style issue, could you write it as `alpha > 0` (variables on left side). Thanks.

Reply:
Thanks for pointing it out. Will correct it.